research-software-directory/RSD-as-a-service
# Scrape mentions doi

# Scraper for mentions

Changes proposed in this pull request:

* Add a scraper that, every 6 minutes, takes mentions from the database with a DOI that were scraped the longest ago, and scrapes its metadata from Crossref or DataCite
* Disable the existing Zotero scraper

How to test:
* Copy over the changes from `.env.example` to `.env` and provide values
* The database structure has changed, clear and build:
	* `docker-compose down --volumes && docker-compose build --parallel && docker-compose up`
* The data-migration script has also changed, build and run:
	* `cd data-migration && docker-compose build && docker-compose up`
* Scraping mentions should work, you can speed it up by running `docker-compose exec scrapers java -cp /usr/myjava/scrapers.jar nl.esciencecenter.rsd.scraper.doi.MainMentions` a few times
* When scraping starts and stops, you should see the messages `Start scraping mentions` and `Done scraping mentions` respectively, hopefully without any error messages in between
* Check (`SELECT * FROM mention ORDER BY scraped_at DESC NULLS LAST;`) that data has indeed been scraped


Closes #298

PR Checklist:
